From ebf33742f4c7bfe55564da429666bc3efd3e78eb Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Wed, 22 Mar 2017 10:19:57 +0800
Subject: [PATCH 1613/5242] MLK-14501 ARM: imx: mu: pass real dev id when
 request shared irq

commit  568993c06ed7a6623c2bea6aae5e25fb42b29839 from
https://source.codeaurora.org/external/imx/linux-imx.git

Pass the real dev id when the shared irq is request.
Otherwise, the request_irq would be failed.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/mu.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/mu.c b/arch/arm/mach-imx/mu.c
index f8786c2..4ab7ef2 100644
--- a/arch/arm/mach-imx/mu.c
+++ b/arch/arm/mach-imx/mu.c
@@ -332,6 +332,7 @@ static int imx_mu_probe(struct platform_device *pdev)
 	int ret;
 	u32 irq;
 	struct device_node *np;
+	struct device *dev = &pdev->dev;
 
 	np = of_find_compatible_node(NULL, NULL, "fsl,imx6sx-mu");
 	mu_base = of_iomap(np, 0);
@@ -343,7 +344,7 @@ static int imx_mu_probe(struct platform_device *pdev)
 	else
 		irq = platform_get_irq(pdev, 0);
 	ret = request_irq(irq, imx_mu_isr,
-			  IRQF_EARLY_RESUME | IRQF_SHARED, "imx-mu", NULL);
+			  IRQF_EARLY_RESUME | IRQF_SHARED, "imx-mu", dev);
 	if (ret) {
 		pr_err("%s: register interrupt %d failed, rc %d\n",
 			__func__, irq, ret);
-- 
1.7.9.5

